pkg/registry/core/service/ipallocator/ipallocator.go

```diff
@@ -0,0 +1,480 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipallocator
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"net/netip"
+	"time"
+
+	networkingv1alpha1 "k8s.io/api/networking/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	networkingv1alpha1informers "k8s.io/client-go/informers/networking/v1alpha1"
+	networkingv1alpha1client "k8s.io/client-go/kubernetes/typed/networking/v1alpha1"
+	networkingv1alpha1listers "k8s.io/client-go/listers/networking/v1alpha1"
+	"k8s.io/client-go/tools/cache"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	netutils "k8s.io/utils/net"
+)
+
+const ControllerName = "ipallocator.k8s.io"
+
+// Allocator implements current ipallocator interface using IPAddress API object
+// and an informer as backend.
+type Allocator struct {
+	net          *net.IPNet
+	firstAddress netip.Addr   // first IP address within the range
+	lastAddress  netip.Addr   // last IP address within the range
+	family       api.IPFamily // family is the IP family of this range
+	max          int64        // cap the total number of IPs available to maxInt64
```

Comment:
yeah, I think I should be carrying on some of this inconsistencies in the code, I have to add a test or something ...

Reply:
> netutils.RangeSize to uint64

this is an incompatible change :/

Reply:
filed https://github.com/kubernetes/utils/pull/277 , but will keep a local function here to not depend on that change ... since this code is gated as alpha this will not be a problem